nixos/tests/common/letsencrypt/common.nix

```diff
@@ -1,12 +1,17 @@
-{ lib, nodes, ... }: {
+{ lib, nodes, pkgs, ... }: let
+  letsencrypt-ca = nodes.letsencrypt.config.test-support.letsencrypt.caCert;
+in {
   networking.nameservers = [
     nodes.letsencrypt.config.networking.primaryIPAddress
   ];
 
+  security.pki.certificateFiles = [ letsencrypt-ca ];
+
   nixpkgs.overlays = lib.singleton (self: super: {
-    cacert = super.cacert.overrideDerivation (drv: {
+    cacert-le = super.cacert.overrideDerivation (drv: {
```

Comment:
```suggestion
    cacert-le = super.cacert-le.overrideDerivation (drv: {
```

Reply:
While this does trim down the rebuilds, the reason I made an override on `cacert` in the first place was to make sure that we override as many variants as possible (eg. all packages that use `cacert` in any way) because the module is designed to be generally used not only for `tests/acme.nix`, that's why the module is in `common/`.

Reply:
To elaborate more on that: It's designed in a way to be used as a drop-in module you can use for writing full-network tests that should be as close as possible to the real world, eg. you just throw in `common/resolver.nix` and `common/letsencrypt` and can use the production config of your deployment for testing, without the need to patch in snakeoil certs.

For example, in a few deployments I don't use the ACME module from NixOS, because I need DNS-01 challenges and so can test the implementation by reusing the modules from `tests/common` without doing a lot of extra patching.

Btw., another option here to avoid the rebuilds could be `system.replaceRuntimeDependencies`.

Reply:
@aszlig while a lot of packages transitively depend on `cacert` during build time, they normally don't bake the certificate store inside their binaries, but we point them to look into `/etc/ssl/certs/ca-(certificates,bundle).crt` at runtime. Custom CAs can be added there through the `security.pki.certificate[Files]` options, which should not rebuild the world, and I'd consider a software not honoring this location a bug.

Reply:
I can update the PR to use that later today!

Thanks for the help @fpletz btw, it's very much appreciated :)

Reply:
@aszlig I agree in principle but as far as I understood pebble's code, you can't provide a CA to be used to issue certificates but rather pebble seems to generate its own temporary CA which can be fetched via HTTP after it's started. So we don't know the CA beforehand and thus we can't even override `pkgs.cacert` or amend `/etc/ssl/certs/ca-certs.crt`.

`pythonPackages.certifi` is still overridden because it has it's own CA trust store (I agree with @flokli that this is technically a bug) that is used by `simp_le`. The snakeoil CA and cert is only used to verify requests to `acme-v02.api.letsencrypt.org` but nothing else.

Reply:
@fpletz: Hm, maybe it makes sense to stick with Boulder then :-/ Or alternatively patch Pebble to use a pre-generated root/intermediate.

@NinjaTrappeur: If you want I can look into getting Boulder to work tonight.

Reply:
@aszlig The tests now pass, we download the certificates from pebble and manually specify them while running curl. While this requires some custom setup in the `testScript`, I think it's still wayy less complicated compared to setting up boulder (or patching pebble to use a pre-generated CA - it's only kept in memory currently)

I'm a bit worried about boulder being a bit too complicated just for a NixOS test - if it would be a proper NixOS module, with someone actually using it in a production setting, that would be another thing, but at least for now, I'd prefer sticking with manually fetching and setting `--cacert` instead of having a large amount of test harness code.

Are you fine with that?

Reply:
@flokli: I think getting in ACMEv2 support is much more important right now rather than being able to use the test module for full-network tests, given that I might (we don't have metrics, so just a guess) be the only one actually using it.

So yes, I'm fine with that. We can still fix the full-network case later.